api/src/main/java/org/apache/cloudstack/api/command/user/vm/DeployVMCmd.java

```diff
@@ -256,12 +281,35 @@ public Long getDomainId() {
                 }
             }
         }
+        if(getBootType() != null){ // export to get
+            if(getBootType() == ApiConstants.BootType.UEFI) {
+                customparameterMap.put(getBootType().toString(), getBootMode().toString());
```

Comment:
@pavanaravapalli cc @DaanHoogland @andrijapanicsb 
What is one of the methods return null?